### PR TITLE
fix: Undo custom `Margin` on `Slider` `Thumb`

### LINKED ISSUE #16484: `Thumb` style uses non-WinUI Margin
### Current behavior

In `Slider_themeresources` `Thumb` uses `-2` as `Margin` on WinUI, whereas we use `0` because of clipping issues. It should be validated if these are still problematic or the workaround can be removed.

Could be related to https://github.com/unoplatform/uno/issues/15449

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### LINKED ISSUE #16900: `Slider`'s track has a margin around the `Thumb`
### Current behavior

The slider's track has a curious gap next to the thumb.
 
image

### Expected behavior

Should not be there, same as on WinUI

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #16484, closes #16900

## PR Type

What kind of change does this PR introduce?

- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Comment by MartinZikmund** [comment]
image

**Comment by MartinZikmund** [comment]
Verified on uno targets and the change does not seem to cause any visual issues anymore

File: src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/Slider_themeresources.xaml [hunk 1]
```diff
@@ -198,8 +198,7 @@
                                         <ControlTemplate TargetType="Thumb">
                                             <!-- Uno specific (LinearGradientBrush borders): We need additional Grid here #6457 -->
                                             <Grid>
-                                                <!-- Uno specific: WinUI uses Margin -2 here, which causes clipping for us. #16484 -->
-                                                <Border x:Name="ThumbBorder" Margin="0" Background="{ThemeResource SliderOuterThumbBackground}" BorderBrush="{ThemeResource ControlStrokeColorDefaultBrush}" BorderThickness="{TemplateBinding BorderThickness}" CornerRadius="{ThemeResource SliderThumbCornerRadius}">
+                                                <Border x:Name="ThumbBorder" Margin="-2" Background="{ThemeResource SliderOuterThumbBackground}" BorderBrush="{ThemeResource ControlStrokeColorDefaultBrush}" BorderThickness="{TemplateBinding BorderThickness}" CornerRadius="{ThemeResource SliderThumbCornerRadius}">
```
Comment (Youssef1313) [inline src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/Slider_themeresources.xaml:201]: Have you confirmed whether Android and iOS work with this?

Comment (MartinZikmund) [reply src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/Slider_themeresources.xaml:201]: Yes, everything seems to be fine 🤔

**Comment by MartinZikmund** [comment]
Verified samples app, and screenshot comparisons don't show anything major

**Pull request merged**